### PR TITLE
Added JS tests for yii.gridView.js (#12840)

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes, it contains some bug fixes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #12840, #12836, #13231, #13232
| Other tested issues / PRs  |  #10302 (https://github.com/yiisoft/yii2/pull/10284#issuecomment-160957558), #10284 (https://github.com/yiisoft/yii2/issues/10217#issuecomment-160594457), #11729

### CODE REVIEW

**Review by samdark — APPROVED** [review]

File: framework/assets/yii.gridView.js [hunk 1]
```diff
@@ -16,7 +16,7 @@
         } else if (typeof method === 'object' || !method) {
             return methods.init.apply(this, arguments);
         } else {
-            $.error('Method ' + method + ' does not exist on jQuery.yiiGridView');
+            $.error('Method ' + method + ' does not exist in jQuery.yiiGridView');
```
Comment (arogachev) [inline framework/assets/yii.gridView.js]: Small grammar nitpick.

File: framework/assets/yii.gridView.js [hunk 2]
```diff
@@ -59,35 +85,38 @@
...
 
-                        return false;
-                    });
+                    return false;
+                });
             });
         },
 
         applyFilter: function () {
-            var $grid = $(this), event;
```
Comment (arogachev) [inline framework/assets/yii.gridView.js]: Declaration of `event` variable is unnecessary here, it's used below in one specific place.

**Review by SilverFire — APPROVED** [review]

File: framework/assets/yii.gridView.js [hunk 3]
```diff
@@ -154,17 +183,20 @@
             if (gridData[id] === undefined) {
                 gridData[id] = {};
             }
+            if (gridEventHandlers[id] === undefined) {
+                gridEventHandlers[id] = {};
+            }
```
Comment (cebe) [inline framework/assets/yii.gridView.js]: should this check be moved to `initEventHandler`? sounds logical to me as it is done always before calling `initEventHandler`.

**Review by cebe — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thank you very very much for these.

**Pull request merged**